[Bug]: UpdateParseWebhook Fails with 404

### Prerequisites

- [x] I have searched existing issues to ensure this bug has not been reported before
- [x] I have read the [troubleshooting guide](https://github.com/arslanbekov/terraform-provider-sendgrid/blob/master/docs/troubleshooting.md)
- [x] I am using a supported version of the provider (v2.x)

### Bug Description

UpdateParseWebhook in the SDK uses the PUT HTTP method, but the SendGrid API only supports PATCH for the /v3/user/webhooks/parse/settings/{hostname} endpoint. This causes any in-place update to a parse webhook to fail with a 404. The issue surfaces most commonly after importing an existing parse webhook via terraform import and then running terraform apply, which triggers an update for any attribute differences (e.g., webhook_security_policy_id, spam_check, send_raw).
Additionally, ReadParseWebhook hardcodes http.StatusInternalServerError for all API errors, discarding the actual HTTP status code. This prevents the resource Read function from distinguishing a 404 (resource deleted) from other failures, causing hard errors instead of graceful state removal.

### Expected Behavior

After importing a parse webhook with terraform import sendgrid_parse_webhook.<name> <hostname>, running terraform apply should successfully update any differing non-ForceNew attributes (spam_check, send_raw, webhook_security_policy_id) via the SendGrid API.

### Actual Behavior

terraform apply fails with an HTTP 404 error on the update operation. The SendGrid API does not recognize the PUT method on the parse webhook settings endpoint and returns a plain-text 404 page not found response.

### Steps to Reproduce

1. Create a parse webhook in SendGrid (via dashboard or API)
2. Define the corresponding sendgrid_parse_webhook resource in Terraform config
3. Run terraform import sendgrid_parse_webhook.<name> <hostname>
4. Run terraform apply (any non-ForceNew attribute difference will trigger an update)
5. Observe the 404 error

### Terraform Configuration

```hcl
resource "sendgrid_webhook_security_policy" "agent" {
  name = "Email agent security policy"
  oauth {
    client_id     = "your-client-id"
    client_secret = "your-client-secret"
    token_url     = "https://example.com/oauth/token"
    scopes        = []
  }
  signature {
    enabled = true
  }
}
resource "sendgrid_parse_webhook" "agent" {
  hostname                   = "agent.example.com"
  url                        = "https://example.com/inbound"
  spam_check                 = true
  send_raw                   = false
  webhook_security_policy_id = sendgrid_webhook_security_policy.agent.id
}
```

### Error Messages and Logs

```shell
2026-03-10T14:00:28.319-0400 [INFO]  provider.terraform-provider-sendgrid: [DEBUG] Waiting for state to become: [success]
2026-03-10T14:00:28.475-0400 [ERROR] provider.terraform-provider-sendgrid: Response contains error diagnostic:
  diagnostic_severity=ERROR
  diagnostic_summary=
  | request failed: resource not found. It may have been deleted outside of Terraform or the ID is incorrect
  |
  | Original error: api response: HTTP 404: 404 page not found
  tf_resource_type=sendgrid_parse_webhook
  tf_rpc=ApplyResourceChange
2026-03-10T14:00:28.485-0400 [ERROR] vertex "sendgrid_parse_webhook.agent" error: request failed: resource not found. It may have been deleted outside of Terraform or the ID is incorrect
```

### Provider Version

v2.1.2

### Terraform Version

1.8.5

### Operating System

Mac OS

### Affected Resource(s)

sendgrid_parse_webhook

### Additional Context

_No response_

### Workaround

- [ ] I have found a workaround (please describe in Additional Context)

### References

_No response_

--- a/sdk/parse_webhook.go
+++ b/sdk/parse_webhook.go
@@ -84,10 +84,10 @@ func (c *Client) ReadParseWebhook(ctx context.Context, hostname string) (*ParseW
 		}
 	}
 
-	respBody, _, err := c.Get(ctx, "GET", "/user/webhooks/parse/settings/"+hostname)
+	respBody, statusCode, err := c.Get(ctx, "GET", "/user/webhooks/parse/settings/"+hostname)
 	if err != nil {
 		return nil, RequestError{
-			StatusCode: http.StatusInternalServerError,
+			StatusCode: statusCode,
 			Err:        err,
 		}
 	}
@@ -109,7 +109,7 @@ func (c *Client) UpdateParseWebhook(ctx context.Context, hostname string, spamCh
 	t.SendRaw = sendRaw
 	t.SecurityPolicy = securityPolicy
 
-	_, _, err := c.Post(ctx, "PUT", "/user/webhooks/parse/settings/"+hostname, t)
+	_, _, err := c.Post(ctx, "PATCH", "/user/webhooks/parse/settings/"+hostname, t)
 	if err != nil {
 		return RequestError{
 			StatusCode: http.StatusInternalServerError,

--- a/sendgrid/resource_sendgrid_parse_webhook.go
+++ b/sendgrid/resource_sendgrid_parse_webhook.go
@@ -21,6 +21,7 @@ package sendgrid
 
 import (
 	"context"
+	"net/http"
 
 	sendgrid "github.com/arslanbekov/terraform-provider-sendgrid/sdk"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
@@ -117,6 +118,11 @@ func resourceSendgridParseWebhookRead(ctx context.Context, d *schema.ResourceDat
 
 	webhook, err := c.ReadParseWebhook(ctx, d.Id())
 	if err.Err != nil {
+		if err.StatusCode == http.StatusNotFound {
+			d.SetId("")
+			return nil
+		}
+
 		return diag.FromErr(err.Err)
 	}
 

--- a/sendgrid/resource_sendgrid_parse_webhook_test.go
+++ b/sendgrid/resource_sendgrid_parse_webhook_test.go
@@ -29,6 +29,12 @@ func TestAccSendgridParseWebhookBasic(t *testing.T) {
 					resource.TestCheckResourceAttr("sendgrid_parse_webhook.test", "spam_check", "true"),
 				),
 			},
+			// Import test
+			{
+				ResourceName:      "sendgrid_parse_webhook.test",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
 		},
 	})
 }
@@ -78,6 +84,37 @@ func TestAccSendgridParseWebhookUpdate(t *testing.T) {
 				Check: resource.ComposeTestCheckFunc(
 					testAccCheckSendgridParseWebhookExists("sendgrid_parse_webhook.test"),
 					resource.TestCheckResourceAttr("sendgrid_parse_webhook.test", "url", urlUpdated),
+				),
+			},
+		},
+	})
+}
+
+func TestAccSendgridParseWebhookUpdateInPlace(t *testing.T) {
+	hostname := "parse-inplace-" + acctest.RandString(10) + ".example.com"
+	url := "https://inplace-" + acctest.RandString(10) + ".com/parse"
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckSendgridParseWebhookDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckSendgridParseWebhookConfigBasic(hostname, url),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckSendgridParseWebhookExists("sendgrid_parse_webhook.test"),
+					resource.TestCheckResourceAttr("sendgrid_parse_webhook.test", "spam_check", "true"),
+					resource.TestCheckResourceAttr("sendgrid_parse_webhook.test", "send_raw", "false"),
+				),
+			},
+			{
+				Config: testAccCheckSendgridParseWebhookConfigUpdatedFields(hostname, url),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckSendgridParseWebhookExists("sendgrid_parse_webhook.test"),
+					resource.TestCheckResourceAttr("sendgrid_parse_webhook.test", "hostname", hostname),
+					resource.TestCheckResourceAttr("sendgrid_parse_webhook.test", "url", url),
+					resource.TestCheckResourceAttr("sendgrid_parse_webhook.test", "spam_check", "false"),
+					resource.TestCheckResourceAttr("sendgrid_parse_webhook.test", "send_raw", "true"),
 				),
 			},
 		},
@@ -152,6 +189,17 @@ resource "sendgrid_parse_webhook" "test" {
 	url        = "%s"
 	spam_check = true
 	send_raw   = false
+}
+`, hostname, url)
+}
+
+func testAccCheckSendgridParseWebhookConfigUpdatedFields(hostname, url string) string {
+	return fmt.Sprintf(`
+resource "sendgrid_parse_webhook" "test" {
+	hostname   = "%s"
+	url        = "%s"
+	spam_check = false
+	send_raw   = true
 }
 `, hostname, url)
 }

